model/utils.go

```diff
@@ -207,6 +207,9 @@ type AppError struct {
 }
 
 func (er *AppError) Error() string {
+	if er == nil {
+		return ""
+	}
```

Comment:
This is not ideal. The nil check should be done at the caller level, not here. We are also working on a vet check to detect nil dereferences and prevent scenarios like this.